pkg/app/appserver/proc.go

```diff
@@ -34,7 +34,8 @@ type Proc struct {
 }
 
 // NewProc constructs `Proc`.
-func NewProc(log *logging.Logger, c appcommon.Config, args []string, stdout, stderr io.Writer) (*Proc, error) {
+func NewProc(log *logging.Logger, c appcommon.Config, args []string, envs map[string]string,
```

Comment:
I think we should put these args into a struct as they got large.

Reply:
@nkryuchkov I've seen @evanlinjin changed this signature in one of his PRs too. I think it's simpler to merge it as is and raise a separate issue to simplify this part. which could be tackled right after this PR and Evan's are merged